tests/test_router.py

```diff
@@ -1,48 +1,49 @@
 # coding=utf-8
 import json
+from typing import Callable
 
 from lbz.misc import NestedDict
 from lbz.router import Router
 from lbz.router import add_route
 
 
 class TestRouter:
-    def setup_method(self):
+    def setup_method(self) -> None:
         # pylint: disable= attribute-defined-outside-init
         self.router = Router()
         self.router.add_route("/", "GET", "x")
 
-    def teardown_method(self, _test_method):
+    def teardown_method(self, _test_method: Callable) -> None:
```

Comment:
I think you could use [types.MethodType](https://docs.python.org/3/library/types.html#types.MethodType) here instead 😉 

_Side note:_ we don't have `mypy` enabled for tests, so they are not checked automatically - introducing types into this big pull request may make a review longer 😐